check_source.py

```diff
@@ -215,8 +215,10 @@ def check_action_delete(self, request, action):
         except urllib2.HTTPError:
             return None
 
-        # Decline the delete request against linked package.
-        links = root.findall('linked')
+        for si in root.findall('sourceinfo'):
+            # Decline the delete request against linked package.
+            links = si.findall('linked')
```

Comment:
Would `links = si.findall('sourceinfo/linked')`work? Based on your xml example it seems like it would, but the previous code placed everything inside of the loop.

982bd86c6ba1b724f4266419aee54370a3c5f24a